samples/zigbee/template/src/main.c

```diff
@@ -0,0 +1,238 @@
+		ZVUNUSED(zb_err_code);
+
+		dk_set_led(IDENTIFY_LED, 0);
+	}
+}
+
+/**@breif Starts identifying the device.
+ *
+ * @param  bufid  Unused parameter, required by ZBOSS scheduler API.
+ */
+static void start_identifying(zb_bufid_t bufid)
+{
+	zb_ret_t zb_err_code;
+
+	ZVUNUSED(bufid);
+
+	/* Check if endpoint is in identifying mode,
+	 * if not put desired endpoint in identifying mode.
+	 */
+	if (dev_ctx.identify_attr.identify_time ==
+	    ZB_ZCL_IDENTIFY_IDENTIFY_TIME_DEFAULT_VALUE) {
+		LOG_INF("Enter identify mode");
+		zb_err_code = zb_bdb_finding_binding_target(
+			APP_TEMPLATE_ENDPOINT);
+		ZB_ERROR_CHECK(zb_err_code);
+	} else {
+		LOG_INF("Cancel identify mode");
+		zb_bdb_finding_binding_target_cancel();
+	}
+}
+
+/**@brief Callback for button events.
+ *
+ * @param[in]   button_state  Bitmask containing buttons state.
+ * @param[in]   has_changed   Bitmask containing buttons
+ *                            that have changed their state.
+ */
+static void button_changed(uint32_t button_state, uint32_t has_changed)
+{
+	/* Calculate bitmask of buttons that are pressed
+	 * and have changed their state.
+	 */
+	uint32_t buttons = button_state & has_changed;
+
+	if (buttons & IDENTIFY_MODE_BUTTON) {
+		ZB_SCHEDULE_APP_CALLBACK(start_identifying, 0);
+	}
+}
+
+/**@brief Function for initializing LEDs and Buttons. */
+static void configure_gpio(void)
+{
+	int err;
+
+	err = dk_buttons_init(button_changed);
+	if (err) {
+		LOG_ERR("Cannot init buttons (err: %d)", err);
+	}
+
+	err = dk_leds_init();
+	if (err) {
+		LOG_ERR("Cannot init LEDs (err: %d)", err);
+	}
+}
+
+
+/**@brief Zigbee stack event handler.
+ *
+ * @param[in]   bufid   Reference to the Zigbee stack buffer
+ *                      used to pass signal.
+ */
+void zboss_signal_handler(zb_bufid_t bufid)
+{
+	/* Update network status LED. */
+	zigbee_led_status_update(bufid, ZIGBEE_NETWORK_STATE_LED);
+
+	/* No application-specific behavior is required.
+	 * Call default signal handler.
+	 */
+	ZB_ERROR_CHECK(zigbee_default_signal_handler(bufid));
+
+	/* All callbacks should either reuse or free passed buffers.
+	 * If bufid == 0, the buffer is invalid (not passed).
+	 */
+	if (bufid) {
+		zb_buf_free(bufid);
+	}
+}
+
+void error(void)
+{
+	dk_set_leds_state(DK_ALL_LEDS_MSK, DK_NO_LEDS_MSK);
+
+	while (true) {
+		/* Spin forever */
+		k_sleep(K_MSEC(1000));
+	}
+}
+
+void main(void)
+{
+	LOG_INF("Starting Zigbee application template example");
+
+	/* Initialize */
+	configure_gpio();
+
+	/* Register device context (endpoints). */
+	ZB_AF_REGISTER_DEVICE_CTX(&app_template_ctx);
+
+	app_clusters_attr_init();
+
+	/* Register handlers to identify notifications */
+	ZB_AF_SET_IDENTIFY_NOTIFICATION_HANDLER(APP_TEMPLATE_ENDPOINT, identify_cb);
+
+	/* Start Zigbee default thread */
+	zigbee_enable();
+
+	LOG_INF("Zigbee application template started");
+
+	while (true) {
+		/* Spin forever */
+		k_sleep(K_FOREVER);
```

Comment:
You can do this, but I just wonder - why not simply exit the thread? It shall not impact other thread's operation.